creusot-contracts/src/ptr_own.rs

```diff
@@ -16,32 +16,42 @@ pub type RawPtr<T> = *const T;
 /// corresponding to the pointer because it is a ghost value. One must thus
 /// remember to explicitly call [`drop`] in order to free the memory tracked by a
 /// `PtrOwn` token.
-#[trusted]
-pub struct PtrOwn<T: ?Sized>(std::marker::PhantomData<T>);
+///
+/// # Safety
+///
+/// When using Creusot to verify the code, all methods should be safe to call. Indeed,
+/// Creusot ensures that every operation on the inner value uses the right [`PtrOwn`] object
+/// created by [`PtrOwn::new`], ensuring safety in a manner similar to [ghost_cell](https://docs.rs/ghost-cell/latest/ghost_cell/).
+pub struct PtrOwn<T: ?Sized> {
+    #[cfg(creusot)]
+    ptr: RawPtr<T>,
+    #[cfg(creusot)]
+    val: Box<T>,
+    #[cfg(not(creusot))]
+    _val: std::marker::PhantomData<T>,
```

Comment:
Why do we need a different implementation in the case we compile without Creusot?